torch/_dynamo/utils.py

```diff
@@ -1061,54 +1105,32 @@ def safe_str(item: Any) -> str:
         remote_cache_version = None
 
-    # Populate the compile_id from the metrics context if it's set. Otherwise
-    # look for it in the compile context.
-    compile_id = metrics.get("compile_id")
-    if not compile_id:
-        compile_id = torch._guards.CompileContext.current_compile_id()
-
     common_metrics = {
-        "compile_id": str(compile_id) if compile_id else None,
         "start_time_us": start_time_ns // 1000,
         "end_time_us": end_time_ns // 1000,
         "duration_us": (end_time_ns - start_time_ns) // 1000,
         "fail_type": exc_type.__qualname__ if exc_type else None,
         "fail_reason": str(exc_value) if exc_value else None,
-        "structured_logging_overhead_us": to_int_us(structured_logging_overhead_s),
+        "structured_logging_overhead_us": to_int_us(
+            torch._logging.get_structured_logging_overhead()
+        ),
         "inductor_config": _scrubbed_inductor_config_for_logging(),
         "cuda_version": torch.version.cuda,
         "triton_version": triton.__version__ if has_triton() else "",
-        "inductor_fx_remote_cache_hit_keys": _convert_collection_to_str(
-            "inductor_fx_remote_cache_hit_keys"
-        ),
-        "inductor_fx_remote_cache_miss_keys": _convert_collection_to_str(
-            "inductor_fx_remote_cache_miss_keys"
-        ),
         "remote_cache_version": remote_cache_version,
         "inductor_fx_remote_cache_backend_type": inductor_fx_remote_cache_backend_type,
     }
 
-    # TODO: The following are legacy fields, populated from the fields that replace
-    # them. Remove these when we decide we can really deprecate them.
-    legacy_metrics = {
-        "start_time": start_time_ns / 1e9,
-        "entire_frame_compile_time_s": us_to_s("dynamo_cumulative_compile_time_us"),
-        "backend_compile_time_s": us_to_s("aot_autograd_cumulative_compile_time_us"),
-        "inductor_compile_time_s": us_to_s("inductor_cumulative_compile_time_us"),
-        "code_gen_time_s": us_to_s("inductor_code_gen_cumulative_compile_time_us"),
-        "remote_cache_time_saved_s": us_to_s("distributed_ephemeral_timeout_us"),
-        "remote_fx_graph_cache_get_time_ms": us_to_ms(
-            "remote_fx_graph_cache_get_time_us"
-        ),
-        "remote_fx_graph_cache_put_time_ms": us_to_ms(
-            "remote_fx_graph_cache_put_time_us"
-        ),
-        "structured_logging_overhead_s": structured_logging_overhead_s,
-    }
+    all_metrics = {**common_metrics, **metrics}
 
-    compilation_metrics = CompilationMetrics(
-        **{**legacy_metrics, **common_metrics, **metrics}
-    )
+    # Populate the compile_id from the metrics context if it's set. Otherwise,
+    # look for it in the current compile context.
+    compile_id = metrics.get("compile_id")
+    if not compile_id:
+        compile_id = torch._guards.CompileContext.current_compile_id()
+    all_metrics["compile_id"] = str(compile_id) if compile_id else None
```

Comment:
Can we add a set_compile_id() method to CompilationMetrics, the new CompilationMetrics is so beautiful with the factory method and all, this adhoc key assignment sticks out.

Reply:
Ok, I know how to fix this